src/Symfony/Contracts/Cache/CacheTrait.php

```diff
@@ -72,6 +72,11 @@ private function doGet(CacheItemPoolInterface $pool, string $key, callable $call
             $item->set($callback($item, $save));
             if ($save) {
                 $pool->save($item);
+            } else {
+                $value = $item->get();
+                if ($value instanceof \Closure) {
```

Comment:
It's related to this PR since this PR is all about reliability and data consistency :)
but I agree that this feature is a big extension and requires substantive discussion.
I'll postpone it for the follow-up PR.